### PR TITLE
Improve documentation distributed scheduler

### PR DESCRIPTION
**Pull request by ericpre**
Try to improve the documentation based on the discussion in https://github.com/hyperspy/rosettasciio/issues/266#issuecomment-2137867084.

### Progress of the PR
- [x] Improve documentation dask scheduler,
- [x] Fix links in documentation,
- [x] Fix error message to point to RosettaSciIO to report error when reading file, 
- [n/a] update docstring (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [n/a] add tests,
- [x] ready for review.

### CODE REVIEW

**Comment by ericpre** [comment]
@CSSFrancis, can you please have a quick look at this PR, please? This should be straightforward to review.

**Comment by CSSFrancis** [comment]
@ericpre Sure I'll do that this morning! Sorry this one kind of got forgotten :)

File: doc/user_guide/big_data.rst [hunk 1]
```diff
@@ -468,15 +468,21 @@ compute the result of all functions that are affected by the axes
...
 
-.. _dask_backends:
+.. _dask_scheduler:
 
-Dask Backends
--------------
+Dask Scheduler
+--------------
 
 Dask is a flexible library for parallel computing in Python. All of the lazy operations in
```
Comment (CSSFrancis) [inline doc/user_guide/big_data.rst]:
```suggestion
Dask is a flexible library for parallel computing in Python. All of the lazy operations in
```
```suggestion
Dask is a flexible library for parallel computing in Python. All of the lazy operations (and many of the non lazy operations) in
```

File: doc/user_guide/big_data.rst [hunk 2]
```diff
@@ -468,15 +468,21 @@ compute the result of all functions that are affected by the axes
...
 hyperspy run through dask. Dask can be used to run computations on a single machine or
-scaled to a cluster. The following example shows how to use dask to run computations on a
-variety of different hardware:
+scaled to a cluster. This section introduce the different scheduler and how to use them
+in HyperSpy - for more details, see the dask documention on
+`scheduling <https://docs.dask.org/en/stable/scheduling.html>`_.
```
Comment (CSSFrancis) [inline doc/user_guide/big_data.rst]:
```suggestion
scaled to a cluster. This section introduces the different schedulers and how to use them
in HyperSpy - for more details, see the dask documention on
`scheduling <https://docs.dask.org/en/stable/scheduling.html>`_.
```

**Review by CSSFrancis — APPROVED** [review]
Just some small details added.  LGTM though

**Pull request merged**